pkg/monitor/worker.go

```diff
@@ -246,11 +246,19 @@ func (mon *monitor) workOne(ctx context.Context, log *logrus.Entry, doc *api.Ope
 		log.Warnf("no hiveShardConfigs set for shard %d", shard)
 	}
 
+	// Need to add the code here
+	// if hourlyRun and preconfiguredNSG
+	// 	create a new client
+	// 	go run new client monitor
```

Comment:
Why do we need to run preconfigured NSG check as separate monitor from existing cluster monitor?

Reply:
This will be an hourly run instead of every minute like what cluster monitor currently does.

Reply:
Cluster monitor does keep track of hourly run. Here is an example: https://github.com/Azure/ARO-RP/blob/8ffcf691bad8e70182f34550f982e414fccde21f/pkg/monitor/cluster/cluster.go#L140

Reply:
that's what I actually did in my update after the original comment.